OpenEdXMobile/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -311,12 +315,33 @@ public void onException(Exception ex) {
                         return; // Return here to avoid falling back to the generic error handler.
                     }
                 }
-                RegisterActivity.this.showAlertDialog(null, ErrorUtils.getErrorMessage(ex, RegisterActivity.this));
+                // If app version is un-supported
+                if (ex instanceof HttpStatusException &&
+                        ((HttpStatusException) ex).getStatusCode() == HttpStatus.UPGRADE_REQUIRED) {
+                    showAppUpdateDialog();
+                } else {
+                    RegisterActivity.this.showAlertDialog(null, ErrorUtils.getErrorMessage(ex, RegisterActivity.this));
+                }
             }
         };
         task.execute();
     }
 
+    public void showAppUpdateDialog() {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
```

Comment:
We should be using `AlertDialogFragment` instead.

*p.s. The advantage of using a DialogFragment is that all the life cycle of the dialog will be handled for you. You will never get the error 'dialog has leaked...' again. Go to DialogFragment and forget Dialogs.*